recorder/src/test/scala/weco/concepts/recorder/IndexSourceTest.scala

```diff
@@ -0,0 +1,186 @@
+        handleSearch(testDocs)
+    )
+
+    val results = IndexSource[TestDoc](
+      elasticHttpClient = testClient,
+      indexName = "test",
+      pageSize = 100
+    )
+      .runWith(TestSink[TestDoc]())
+      .request(1000)
+      .expectNextN(1000)
+
+    results shouldBe testDocs
+  }
+
+  it("reuses its point-in-time ID") {
+    implicit val actorSystem: ActorSystem = ActorSystem("test")
+    implicit val ec: ExecutionContext = actorSystem.dispatcher
+    val testDocs = (1 to 1000).map(id => TestDoc(id.toString))
+    val testClient = new TestElasticHttpClient(
+      handlePitCreation orElse
+        handleSearch(testDocs)
+    )
+
+    IndexSource[TestDoc](
+      elasticHttpClient = testClient,
+      indexName = "test",
+      pageSize = 100
+    )
+      .runWith(TestSink[TestDoc]())
+      .request(1000)
+      .expectNextN(1000)
+
+    val usedPitIds = Future
+      .sequence(testClient.requests.collect {
+        case HttpRequest(HttpMethods.POST, uri, _, entity, _)
+            if uri.path.endsWith("_search") =>
+          Unmarshal(entity)
+            .to[String]
+            .map(ujson.read(_))
+            .map(_.opt[String]("pit", "id"))
+      })
+      .map(_.flatten)
+
+    whenReady(usedPitIds) { ids =>
+      ids.toSet.size shouldBe 1
+    }
+  }
+
+  it("replaces the point-in-time ID if a search updates it") {
+    implicit val actorSystem: ActorSystem = ActorSystem("test")
+    implicit val ec: ExecutionContext = actorSystem.dispatcher
+    val testDocs = (1 to 200).map(id => TestDoc(id.toString))
+    val testClient = new TestElasticHttpClient(
+      handlePitCreation orElse
+        handleSearch(testDocs, _.reverse) orElse
+        handlePitDeletion
+    )
+
+    IndexSource[TestDoc](
+      elasticHttpClient = testClient,
+      indexName = "test",
+      pageSize = 100
+    )
+      .runWith(TestSink[TestDoc]())
+      .request(200)
+      .expectNextN(200)
+
+    val usedPitIds = Future
+      .sequence(testClient.requests.collect {
+        case HttpRequest(HttpMethods.POST, uri, _, entity, _)
+            if uri.path.endsWith("_search") =>
+          Unmarshal(entity)
+            .to[String]
+            .map(ujson.read(_))
+            .map(_.opt[String]("pit", "id"))
+      })
+      .map(_.flatten)
+
+    whenReady(usedPitIds) { ids =>
+      ids.toSet.size shouldBe 2
+    }
+  }
+
+  it(
+    "deletes the point-in-time and closes itself when no more results are returned"
+  ) {
+    implicit val actorSystem: ActorSystem = ActorSystem("test")
+    implicit val ec: ExecutionContext = actorSystem.dispatcher
+    val testClient = new TestElasticHttpClient(
+      handlePitCreation orElse
+        handleSearch(Nil) orElse
+        handlePitDeletion
+    )
+
+    IndexSource[TestDoc](
+      elasticHttpClient = testClient,
+      indexName = "test",
+      pageSize = 100
+    )
+      .runWith(TestSink[TestDoc]())
+      .ensureSubscription()
+      .expectComplete()
+
+    testClient.requests.find {
```

Comment:
The use of `find` (and `contain` in the initialisation test), ensures that the calls are made, but not necessarily in an appropriate order.

This could pass if the application first deletes the pit, then searches, then creates the pit